web/org.eclipse.xtend.web/src/main/java/org/eclipse/xtend/web/XtendServiceDispatcher.xtend

```diff
@@ -0,0 +1,62 @@
+package org.eclipse.xtend.web
+
+import com.google.inject.Inject
+import com.google.inject.Singleton
+import org.eclipse.xtend.web.server.java2xtend.Java2XtendService
+import org.eclipse.xtext.util.internal.Log
+import org.eclipse.xtext.web.server.IServiceContext
+import org.eclipse.xtext.web.server.InvalidRequestException
+import org.eclipse.xtext.web.server.XtextServiceDispatcher
+import org.eclipse.xtext.web.server.generator.GeneratorService
+
+@Log
+@Singleton
+class XtendServiceDispatcher extends XtextServiceDispatcher {
+	
+	// TODO propose Gerrit to Xtext to make XtextServiceDispatcher's @Inject protected
+	@Inject GeneratorService generatorService
+	
+	@Inject Java2XtendService java2xtendService
+	
+	override protected createServiceDescriptor(String serviceType, IServiceContext context) {
+		switch serviceType {
+			case 'java2xtend':
+				getJava2XtendService(context)
+			default:
+				super.createServiceDescriptor(serviceType, context)	
+		}
+	}
+	
+	/** 
+	 * Xtend may generate several *.java from one *.xtend,
+	 * so we don't have an 'artifact' request parameter and
+	 * instead of generatorService.getArtifact() need to compute().
+	 */
```

Comment:
I think it would be better to create a new service for this instead of changing the semantics of the original generator service.